### ISSUE TITLE
porting: status in p7 branch

### ISSUE THREAD

**Issue by jkeenan** [atoomic/perl#54]
In this ticket we will track our core-p7 progress on tests in directory `porting`.

As 2020-07-09 10:20 EDT, we are getting this output from `make test` for the files in this directory.

```
porting/args_assert.t .............................................. ok
porting/authors.t .................................................. ok
porting/bench.t .................................................... skipped: no valgrind
porting/bench_selftest.t ........................................... ok
porting/bincompat.t ................................................ ok
porting/checkcase.t ................................................ ok
porting/checkcfgvar.t .............................................. ok
# not ok 1 - cpan/Archive-Tar/lib/Archive/Tar.pm version 2.36
# not ok 2 - cpan/Archive-Tar/lib/Archive/Tar/Constant.pm version 2.36
# not ok 3 - cpan/Archive-Tar/lib/Archive/Tar/File.pm version 2.36
# not ok 4 - cpan/CPAN-Meta-YAML/lib/CPAN/Meta/YAML.pm version 0.018
# not ok 5 - cpan/CPAN/lib/CPAN.pm version 2.27
# not ok 6 - cpan/CPAN/lib/CPAN/Author.pm version 5.5002
# not ok 7 - cpan/CPAN/lib/CPAN/Bundle.pm version 5.5005
# not ok 8 - cpan/CPAN/lib/CPAN/Complete.pm version 5.5001
# not ok 9 - cpan/CPAN/lib/CPAN/Distribution.pm version 2.27
# not ok 10 - cpan/CPAN/lib/CPAN/Distroprefs.pm version 6.0001
# not ok 11 - cpan/CPAN/lib/CPAN/Exception/blocked_urllist.pm version 1.001
# not ok 12 - cpan/CPAN/lib/CPAN/FirstTime.pm version 5.5314
# not ok 13 - cpan/CPAN/lib/CPAN/HandleConfig.pm version 5.5011
# not ok 14 - cpan/CPAN/lib/CPAN/Index.pm version 2.12
# not ok 15 - cpan/CPAN/lib/CPAN/InfoObj.pm version 5.5
# not ok 16 - cpan/CPAN/lib/CPAN/Module.pm version 5.5003
# not ok 17 - cpan/CPAN/lib/CPAN/Nox.pm version 5.5001
# not ok 18 - cpan/CPAN/lib/CPAN/Plugin/Specfile.pm version 0.02
# not ok 19 - cpan/CPAN/lib/CPAN/Prompt.pm version 5.5
# not ok 20 - cpan/CPAN/lib/CPAN/Shell.pm version 5.5009
# not ok 21 - cpan/CPAN/lib/CPAN/Tarzip.pm version 5.5013
# not ok 22 - cpan/Compress-Raw-Bzip2/lib/Compress/Raw/Bzip2.pm version 2.093
# not ok 23 - cpan/Compress-Raw-Zlib/lib/Compress/Raw/Zlib.pm version 2.093
# not ok 24 - cpan/DB_File/DB_File.pm version 1.853
# not ok 25 - cpan/Digest-MD5/MD5.pm version 2.55_01
# not ok 26 - cpan/Encode/Encode.pm version 3.06
# not ok 27 - cpan/Encode/encoding.pm version 3.00
# not ok 28 - cpan/Encode/lib/Encode/Alias.pm version 2.24
# not ok 29 - cpan/Encode/lib/Encode/CN/HZ.pm version 2.10
# not ok 30 - cpan/Encode/lib/Encode/Config.pm version 2.05
# not ok 31 - cpan/Encode/lib/Encode/Encoding.pm version 2.08
# not ok 32 - cpan/Encode/lib/Encode/GSM0338.pm version 2.07
# not ok 33 - cpan/Encode/lib/Encode/Guess.pm version 2.08
# not ok 34 - cpan/Encode/lib/Encode/JP/JIS7.pm version 2.08
# not ok 35 - cpan/Encode/lib/Encode/MIME/Header.pm version 2.28
# not ok 36 - cpan/Encode/lib/Encode/Unicode/UTF7.pm version 2.10
# not ok 37 - cpan/ExtUtils-Constant/lib/ExtUtils/Constant.pm version 0.25
# not ok 38 - cpan/ExtUtils-Install/lib/ExtUtils/Install.pm version 2.14
# not ok 39 - cpan/ExtUtils-MakeMaker/lib/ExtUtils/Command/MM.pm version 7.44
# not ok 40 - cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Any.pm version 7.44
# not ok 41 - cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_OS2.pm version 7.44
# not ok 42 - cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_Unix.pm version 7.44
# not ok 43 - cpan/ExtUtils-MakeMaker/lib/ExtUtils/MM_VMS.pm version 7.44
# not ok 44 - cpan/ExtUtils-MakeMaker/lib/ExtUtils/MakeMaker.pm version 7.44
# not ok 45 - cpan/ExtUtils-MakeMaker/lib/ExtUtils/MakeMaker/version.pm version 7.44
# not ok 46 - cpan/ExtUtils-MakeMaker/lib/ExtUtils/Mkbootstrap.pm version 7.44
# not ok 47 - cpan/Getopt-Long/lib/Getopt/Long.pm version 2.51
# not ok 48 - cpan/IO-Compress/lib/Compress/Zlib.pm version 2.093
# not ok 49 - cpan/IO-Compress/lib/IO/Compress/Adapter/Bzip2.pm version 2.093
# not ok 50 - cpan/IO-Compress/lib/IO/Compress/Adapter/Deflate.pm version 2.093
# not ok 51 - cpan/IO-Compress/lib/IO/Compress/Base.pm version 2.093
# not ok 52 - cpan/IO-Compress/lib/IO/Compress/Base/Common.pm version 2.093
# not ok 53 - cpan/IO-Compress/lib/IO/Compress/Bzip2.pm version 2.093
# not ok 54 - cpan/IO-Compress/lib/IO/Compress/Deflate.pm version 2.093
# not ok 55 - cpan/IO-Compress/lib/IO/Compress/Gzip.pm version 2.093
# not ok 56 - cpan/IO-Compress/lib/IO/Compress/RawDeflate.pm version 2.093
# not ok 57 - cpan/IO-Compress/lib/IO/Compress/Zip.pm version 2.093
# not ok 58 - cpan/IO-Compress/lib/IO/Uncompress/Adapter/Bunzip2.pm version 2.093
# not ok 59 - cpan/IO-Compress/lib/IO/Uncompress/Adapter/Identity.pm version 2.093
# not ok 60 - cpan/IO-Compress/lib/IO/Uncompress/Adapter/Inflate.pm version 2.093
# not ok 61 - cpan/IO-Compress/lib/IO/Uncompress/AnyInflate.pm version 2.093
# not ok 62 - cpan/IO-Compress/lib/IO/Uncompress/AnyUncompress.pm version 2.093
# not ok 63 - cpan/IO-Compress/lib/IO/Uncompress/Base.pm version 2.093
# not ok 64 - cpan/IO-Compress/lib/IO/Uncompress/Bunzip2.pm version 2.093
# not ok 65 - cpan/IO-Compress/lib/IO/Uncompress/Gunzip.pm version 2.093
# not ok 66 - cpan/IO-Compress/lib/IO/Uncompress/Inflate.pm version 2.093
# not ok 67 - cpan/IO-Compress/lib/IO/Uncompress/RawInflate.pm version 2.093
# not ok 68 - cpan/IO-Compress/lib/IO/Uncompress/Unzip.pm version 2.093
# not ok 70 - cpan/IO-Socket-IP/lib/IO/Socket/IP.pm version 0.39
# not ok 71 - cpan/IO-Zlib/Zlib.pm version 1.10
# not ok 72 - cpan/JSON-PP/lib/JSON/PP.pm version 4.04
# not ok 73 - cpan/MIME-Base64/Base64.pm version 3.15
# not ok 74 - cpan/MIME-Base64/QuotedPrint.pm version 3.13
# not ok 75 - cpan/Math-BigInt/lib/Math/BigInt.pm version 1.999818
# not ok 76 - cpan/Math-Complex/lib/Math/Complex.pm version 1.5901
# not ok 77 - cpan/Math-Complex/lib/Math/Trig.pm version 1.23
# not ok 78 - cpan/Memoize/Memoize.pm version 1.03_01
# not ok 80 - cpan/Memoize/Memoize/Expire.pm version 1.03
# not ok 81 - cpan/Memoize/Memoize/ExpireFile.pm version 1.03
# not ok 82 - cpan/Memoize/Memoize/ExpireTest.pm version 1.03
# not ok 83 - cpan/Memoize/Memoize/NDBM_File.pm version 1.03
# not ok 84 - cpan/Memoize/Memoize/SDBM_File.pm version 1.03
# not ok 85 - cpan/Memoize/Memoize/Storable.pm version 1.03
# not ok 86 - cpan/Module-Load-Conditional/lib/Module/Load/Conditional.pm version 0.70
# not ok 87 - cpan/Module-Load/lib/Module/Load.pm version 0.34
# not ok 88 - cpan/Module-Loaded/lib/Module/Loaded.pm version 0.08
# not ok 89 - cpan/Module-Metadata/lib/Module/Metadata.pm version 1.000037
# not ok 90 - cpan/PerlIO-via-QuotedPrint/lib/PerlIO/via/QuotedPrint.pm version 0.08
# not ok 91 - cpan/Pod-Simple/lib/Pod/Simple.pm version 3.40
# not ok 92 - cpan/Pod-Simple/lib/Pod/Simple/BlackBox.pm version 3.40
# not ok 93 - cpan/Pod-Simple/lib/Pod/Simple/DumpAsText.pm version 3.40
# not ok 94 - cpan/Pod-Simple/lib/Pod/Simple/DumpAsXML.pm version 3.40
# not ok 95 - cpan/Pod-Simple/lib/Pod/Simple/HTMLLegacy.pm version 5.01
# not ok 96 - cpan/Pod-Simple/lib/Pod/Simple/LinkSection.pm version 3.40
# not ok 97 - cpan/Pod-Simple/lib/Pod/Simple/Progress.pm version 3.40
# not ok 98 - cpan/Pod-Simple/lib/Pod/Simple/PullParser.pm version 3.40
# not ok 99 - cpan/Pod-Simple/lib/Pod/Simple/PullParserToken.pm version 3.40
# not ok 100 - cpan/Pod-Usage/lib/Pod/Usage.pm version 1.69
# not ok 101 - cpan/Pod-Usage/t/inc/Pod/InputObjects.pm version 1.60
# not ok 102 - cpan/Pod-Usage/t/inc/Pod/Parser.pm version 1.60
# not ok 103 - cpan/Pod-Usage/t/inc/Pod/PlainText.pm version 2.06
# not ok 104 - cpan/Pod-Usage/t/inc/Pod/Select.pm version 1.60
# not ok 105 - cpan/Scalar-List-Utils/lib/List/Util.pm version 1.55
# not ok 106 - cpan/Scalar-List-Utils/lib/Scalar/Util.pm version 1.55
# not ok 107 - cpan/Socket/Socket.pm version 2.029
# not ok 108 - cpan/Sys-Syslog/Syslog.pm version 0.36
# not ok 109 - cpan/Test-Harness/lib/TAP/Parser.pm version 3.42
# not ok 110 - cpan/Test-Harness/lib/Test/Harness.pm version 3.42
# not ok 111 - cpan/Test-Simple/lib/Test/Builder/IO/Scalar.pm version 2.114
# not ok 112 - cpan/Test-Simple/lib/Test/More.pm version 1.302175
# not ok 113 - cpan/Test-Simple/lib/Test2/API.pm version 1.302175
# not ok 114 - cpan/Test-Simple/lib/Test2/API/Context.pm version 1.302175
# not ok 115 - cpan/Test-Simple/lib/Test2/Hub.pm version 1.302175
# not ok 116 - cpan/Test-Simple/lib/Test2/Tools/Tiny.pm version 1.302175
# not ok 117 - cpan/Test-Simple/lib/Test2/Util.pm version 1.302175
# not ok 118 - cpan/Test-Simple/lib/Test2/Util/HashBase.pm version 1.302175
# not ok 119 - cpan/Test-Simple/lib/Test2/Util/Trace.pm version 1.302175
# not ok 120 - cpan/Text-Balanced/lib/Text/Balanced.pm version 2.03
# not ok 121 - cpan/Text-Tabs/lib/Text/Tabs.pm version 2013.0523
# not ok 122 - cpan/Text-Tabs/lib/Text/Wrap.pm version 2013.0523
# not ok 123 - cpan/Time-Piece/Piece.pm version 1.3401
# not ok 124 - cpan/Win32/Win32.pm version 0.53
# not ok 125 - cpan/Win32API-File/File.pm version 0.1203_01
# not ok 126 - cpan/autodie/lib/autodie/skip.pm version 2.32
# not ok 127 - cpan/bignum/lib/bigint.pm version 0.51
# not ok 128 - cpan/libnet/lib/Net/Config.pm version 3.11
# not ok 129 - cpan/libnet/lib/Net/FTP.pm version 3.11
# not ok 130 - cpan/libnet/lib/Net/FTP/A.pm version 3.11
# not ok 131 - cpan/podlators/lib/Pod/Text/Termcap.pm version 4.14
# not ok 132 - cpan/version/lib/version.pm version 0.9924
# not ok 133 - dist/Data-Dumper/Dumper.pm version 2.174
# not ok 134 - dist/Devel-SelfStubber/lib/Devel/SelfStubber.pm version 1.06
# not ok 135 - dist/Env/lib/Env.pm version 1.04
# not ok 136 - dist/ExtUtils-CBuilder/lib/ExtUtils/CBuilder/Platform/os2.pm version 0.280234
# not ok 137 - dist/ExtUtils-ParseXS/t/XSTest.pm version 0.01
# not ok 138 - dist/ExtUtils-ParseXS/t/XSUsage.pm version 0.01
# not ok 139 - dist/Filter-Simple/lib/Filter/Simple.pm version 0.96
# not ok 140 - dist/FindBin/lib/FindBin.pm version 1.51
# not ok 152 - dist/Net-Ping/lib/Net/Ping.pm version 2.72
# not ok 153 - dist/Safe/Safe.pm version 2.41
# not ok 154 - dist/Storable/Storable.pm version 3.21
# not ok 155 - dist/Term-ReadLine/lib/Term/ReadLine.pm version 1.17
# not ok 156 - dist/Test/lib/Test.pm version 1.31
# not ok 158 - dist/Unicode-Normalize/Normalize.pm version 1.27
# not ok 160 - dist/base/lib/base.pm version 2.27
# not ok 162 - dist/threads-shared/lib/threads/shared.pm version 1.61
# not ok 163 - dist/threads/lib/threads.pm version 2.25
# not ok 164 - ext/B/B/Concise.pm version 1.004
# not ok 165 - ext/Devel-Peek/Peek.pm version 1.28
# not ok 166 - ext/DynaLoader/DynaLoader_pm.PL version 1.47
# not ok 167 - ext/Errno/Errno_pm.PL version 1.30
# not ok 168 - ext/File-Glob/Glob.pm version 1.33
# not ok 169 - ext/Hash-Util/lib/Hash/Util.pm version 0.23
# not ok 170 - ext/VMS-DCLsym/DCLsym.pm version 1.09
# not ok 171 - ext/VMS-Stdio/Stdio.pm version 2.45
# not ok 172 - ext/XS-APItest/APItest.pm version 1.09
# not ok 173 - ext/attributes/attributes.pm version 0.33
# not ok 174 - ext/re/re.pm version 0.40
# not ok 175 - haiku/Haiku/Haiku.pm version 0.35
# not ok 178 - lib/Benchmark.pm version 1.23
# not ok 179 - lib/English.pm version 1.11
# not ok 182 - lib/Getopt/Std.pm version 1.12
# not ok 183 - lib/Symbol.pm version 1.08
# not ok 187 - lib/Tie/SubstrHash.pm version 1.00
# not ok 188 - lib/Unicode/UCD.pm version 0.75
# not ok 190 - lib/bytes.pm version 1.07
# not ok 191 - lib/deprecate.pm version 0.04
# not ok 200 - lib/sigtrap.pm version 1.09
# not ok 202 - lib/utf8.pm version 1.22
# not ok 203 - lib/warnings.pm version 1.47
# not ok 204 - symbian/ext/Moped/Msg/Msg.pm version 0.01
porting/cmp_version.t .............................................. 
Failed 171/204 subtests 
porting/copyright.t ................................................ ok
porting/corelist.t ................................................. ok
porting/customized.t ............................................... ok
porting/diag.t ..................................................... ok
porting/dual-life.t ................................................ ok
porting/exec-bit.t ................................................. ok
porting/extrefs.t .................................................. ok
porting/filenames.t ................................................ ok
porting/FindExt.t .................................................. ok
porting/globvar.t .................................................. ok
porting/libperl.t .................................................. ok
porting/maintainers.t .............................................. ok
porting/manifest.t ................................................. ok
porting/pending-author.t ........................................... skipped: No pending changes
porting/perlfunc.t ................................................. ok
porting/pod_rules.t ................................................ ok
#   "The NAME should have a dash and short description after it"
#     p7 near line 216 of regen/pX.pl
#     p7 near line 267 of regen/pX.pl
#   "There is no NAME" 
#      near line ??? of regen/pX.pl
# See end of this test output for your options on silencing this
#   "Pod NAME already used" 
#     'lib/p7.pm' also has NAME 'p7' near line ??? of lib/p5.pm
#   "The NAME should have a dash and short description after it" 
#     p7 near line 58 of lib/p5.pm
# See end of this test output for your options on silencing this
#   "Pod NAME already used" 
#     'lib/p5.pm' also has NAME 'p7' near line ??? of lib/p7.pm
# See end of this test output for your options on silencing this
#   "? Should you be using L<...> instead of" 
#     C<DB::sub> near line 2962 of lib/perl7db.pl
# See end of this test output for your options on silencing this
# 
# HOW TO GET porting/podcheck.t TO PASS
# 
# There were 4 files that had new potential problems identified.
# Some of them may be real, and some of them may be false positives because
# this program isn't as smart as it likes to think it is.  You can teach this
# program to ignore the issues it has identified, and hence pass, by doing the
# following:
# 
# 1) If a problem is about a link to an unknown module or man page that
#    you know exists, re-run the command something like:
#       ./perl -I../lib porting/podcheck.t --add_link MODULE man_page ...
#    (MODULEs should look like Foo::Bar, and man_pages should look like
#    bar(3c); don't do this for a module or man page that you aren't sure
#    about; instead treat as another type of issue and follow the
#    instructions below.)
# 
# 2) For other issues, decide if each should be fixed now or not.  Fix the
#    ones you decided to, and rerun this test to verify that the fixes
#    worked.
# 
# 3) If there remain false positive or problems that you don't plan to fix right
#    now,
#    run this test script by hand, using the following formula (on
#    Un*x-like machines):
#         cd t
#         ./perl -I../lib porting/podcheck.t --regen
# 
#    That should cause all current potential problems to be accepted by
#    the program, so that the next time it runs, they won't be flagged.
#    For a few files, such as perltoc, certain issues will always be
#    expected, and more of the same will be added over time.  For those,
#    before you do the regen, you can edit
#    /home/jkeenan/gitwork/perl/t/porting/known_pod_issues.dat
#    and find the entry for the module's file and specific error message,
#    and change the count of known potential problems to -1.
porting/podcheck.t ................................................. 
Failed 5/1445 subtests 
	(less 1057 skipped subtests: 383 okay)
porting/re_context.t ............................................... ok
porting/readme.t ................................................... ok
'lib/p7.pm-new' and 'lib/p7.pm' differ
'perl7.h-new' and 'perl7.h' differ
porting/regen.t .................................................... 
Failed 2/48 subtests 
porting/ss_dup.t ................................................... ok
porting/test_bootstrap.t ........................................... ok
# Failed test 17 - Porting/checkpodencoding.pl compiles at porting/utils.t line 85
#      got "Name \"DB::single\" used only once: possible typo at Porting/checkpodencoding.pl line 64.\nPorting/checkpodencoding.pl syntax OK\n"
# expected "Porting/checkpodencoding.pl syntax OK\n"
# when executing perl with '-c Porting/checkpodencoding.pl'
# Failed test 31 - Porting/git-find-p4-change compiles at porting/utils.t line 85
#      got "Global symbol \"$change\" requires explicit package name (did you forget to declare \"my $change\"?) at Porting/git-find-p4-change line 14.\nGlobal symbol \"$change\" requires explicit package name (did you forget to declare \"my $change\"?) at Porting/git-find-p4-change line 20.\nPorting/git-find-p4-change had compilation errors.\n"
# expected "Porting/git-find-p4-change syntax OK\n"
# when executing perl with '-c Porting/git-find-p4-change'
# Failed test 36 - Porting/Maintainers.pl compiles at porting/utils.t line 85
#      got "Global symbol \"@IGNORABLE\" requires explicit package name (did you forget to declare \"my @IGNORABLE\"?) at Porting/Maintainers.pl line 18.\nGlobal symbol \"%Modules\" requires explicit package name (did you forget to declare \"my %Modules\"?) at Porting/Maintainers.pl line 123.\nGlobal symbol \"%Modules\" requires explicit package name (did you forget to declare \"my %Modules\"?) at Porting/Maintainers.pl line 1437.\nGlobal symbol \"%Modules\" requires explicit package name (did you forget to declare \"my %Modules\"?) at Porting/Maintainers.pl line 1442.\nGlobal symbol \"%Modules\" requires explicit package name (did you forget to declare \"my %Modules\"?) at Porting/Maintainers.pl line 1444.\nGlobal symbol \"%Modules\" requires explicit package name (did you forget to declare \"my %Modules\"?) at Porting/Maintainers.pl line 1446.\nGlobal symbol \"%Modules\" requires explicit package name (did you forget to declare \"my %Modules\"?) at Porting/Maintainers.pl line 1447.\nGlobal symbol \"%Modules\" requires explicit package name (did you forget to declare \"my %Modules\"?) at Porting/Maintainers.pl line 1449.\nGlobal symbol \"%Modules\" requires explicit package name (did you forget to declare \"my %Modules\"?) at Porting/Maintainers.pl line 1450.\nGlobal symbol \"%Modules\" requires explicit package name (did you forget to declare \"my %Modules\"?) at Porting/Maintainers.pl line 1453.\nPorting/Maintainers.pl had compilation errors.\n"
# expected "Porting/Maintainers.pl syntax OK\n"
# when executing perl with '-c Porting/Maintainers.pl'
# Failed test 43 - Porting/makerel compiles at porting/utils.t line 85
#      got "Global symbol \"$relroot\" requires explicit package name (did you forget to declare \"my $relroot\"?) at Porting/makerel line 46.\nGlobal symbol \"$revision\" requires explicit package name (did you forget to declare \"my $revision\"?) at Porting/makerel line 56.\nGlobal symbol \"$patchlevel\" requires explicit package name (did you forget to declare \"my $patchlevel\"?) at Porting/makerel line 57.\nGlobal symbol \"$subversion\" requires explicit package name (did you forget to declare \"my $subversion\"?) at Porting/makerel line 58.\nGlobal symbol \"$subversion\" requires explicit package name (did you forget to declare \"my $subversion\"?) at Porting/makerel line 59.\nGlobal symbol \"$vers\" requires explicit package name (did you forget to declare \"my $vers\"?) at Porting/makerel line 60.\nGlobal symbol \"$revision\" requires explicit package name (did you forget to declare \"my $revision\"?) at Porting/makerel line 60.\nGlobal symbol \"$patchlevel\" requires explicit package name (did you forget to declare \"my $patchlevel\"?) at Porting/makerel line 60.\nGlobal symbol \"$subversion\" requires explicit package name (did you forget to declare \"my $subversion\"?) at Porting/makerel line 60.\nGlobal symbol \"$perl\" requires explicit package name (did you forget to declare \"my $perl\"?) at Porting/makerel line 70.\nGlobal symbol \"$vers\" requires explicit package name (did you forget to declare \"my $vers\"?) at Porting/makerel line 70.\nGlobal symbol \"$reldir\" requires explicit package name (did you forget to declare \"my $reldir\"?) at Porting/makerel line 71.\nGlobal symbol \"$perl\" requires explicit package name (did you forget to declare \"my $perl\"?) at Porting/makerel line 71.\nGlobal symbol \"$reldir\" requires explicit package name (did you forget to declare \"my $reldir\"?) at Porting/makerel line 74.\nGlobal symbol \"$perl\" requires explicit package name (did you forget to declare \"my $perl\"?) at Porting/makerel line 76.\nGlobal symbol \"$relroot\" requires explicit package name (did you forget to declare \"my $relroot\"?) at Porting/makerel line 76.\nGlobal symbol \"$reldir\" requires explicit package name (did you forget to declare \"my $reldir\"?) at Porting/makerel line 76.\nGlobal symbol \"$relroot\" requires explicit package name (did you forget to declare \"my $relroot\"?) at Porting/makerel line 78.\nGlobal symbol \"$reldir\" requires explicit package name (did you forget to declare \"my $reldir\"?) at Porting/makerel line 78.\nGlobal symbol \"$missfile\" requires explicit package name (did you forget to declare \"my $missfile\"?) at Porting/makerel line 81.\nGlobal symbol \"$missentry\" requires explicit package name (did you forget to declare \"my $missentry\"?) at Porting/makerel line 81.\nGlobal symbol \"$missentry\" requires explicit package name (did you forget to declare \"my $missentry\"?) at Porting/makerel line 83.\nGlobal symbol \"$missentry\" requires explicit package name (did you forget to declare \"my $missentry\"?) at Porting/makerel line 83.\nGlobal symbol \"$missfile\" requires explicit package name (did you forget to declare \"my $missfile\"?) at Porting/makerel line 84.\nGlobal symbol \"$missfile\" requires explicit package name (did you forget to declare \"my $missfile\"?) at Porting/makerel line 86.\nGlobal symbol \"$missentry\" requires explicit package name (did you forget to declare \"my $missentry\"?) at Porting/makerel line 88.\nGlobal symbol \"$missentry\" requires explicit package name (did you forget to declare \"my $missentry\"?) at Porting/makerel line 90.\nGlobal symbol \"$missentry\" requires explicit package name (did you forget to declare \"my $missentry\"?) at Porting/makerel line 93.\nGlobal symbol \"$missentry\" requires explicit package name (did you forget to declare \"my $missentry\"?) at Porting/makerel line 100.\nGlobal symbol \"$missfile\" requires explicit package name (did you forget to declare \"my $missfile\"?) at Porting/makerel line 100.\nGlobal symbol \"$relroot\" requires explicit package name (did you forget to declare \"my $relroot\"?) at Porting/makerel line 105.\nGlobal symbol \"$reldir\" requires explicit package name (did you forget to declare \"my $reldir\"?) at Porting/makerel line 105.\nGlobal symbol \"$relroot\" requires explicit package name (did you forget to declare \"my $relroot\"?) at Porting/makerel line 106.\nGlobal symbol \"$reldir\" requires explicit package name (did you forget to declare \"my $reldir\"?) at Porting/makerel line 106.\nGlobal symbol \"$relroot\" requires explicit package name (did you forget to declare \"my $relroot\"?) at Porting/makerel line 106.\nGlobal symbol \"$reldir\" requires explicit package name (did you forget to declare \"my $reldir\"?) at Porting/makerel line 106.\nGlobal symbol \"$relroot\" requires explicit package name (did you forget to declare \"my $relroot\"?) at Porting/makerel line 107.\nGlobal symbol \"$reldir\" requires explicit package name (did you forget to declare \"my $reldir\"?) at Porting/makerel line 107.\nGlobal symbol \"$relroot\" requires explicit package name (did you forget to declare \"my $relroot\"?) at Porting/makerel line 107.\nGlobal symbol \"$reldir\" requires explicit package name (did you forget to declare \"my $reldir\"?) at Porting/makerel line 107.\nGlobal symbol \"$relroot\" requires explicit package name (did you forget to declare \"my $relroot\"?) at Porting/makerel line 108.\nGlobal symbol \"$reldir\" requires explicit package name (did you forget to declare \"my $reldir\"?) at Porting/makerel line 108.\nGlobal symbol \"$relroot\" requires explicit package name (did you forget to declare \"my $relroot\"?) at Porting/makerel line 108.\nGlobal symbol \"$reldir\" requires explicit package name (did you forget to declare \"my $reldir\"?) at Porting/makerel line 108.\nGlobal symbol \"$relroot\" requires explicit package name (did you forget to declare \"my $relroot\"?) at Porting/makerel line 109.\nGlobal symbol \"$reldir\" requires explicit package name (did you forget to declare \"my $reldir\"?) at Porting/makerel line 109.\nGlobal symbol \"$relroot\" requires explicit package name (did you forget to declare \"my $relroot\"?) at Porting/makerel line 109.\nGlobal symbol \"$reldir\" requires explicit package name (did you forget to declare \"my $reldir\"?) at Porting/makerel line 109.\nGlobal symbol \"$cmd\" requires explicit package name (did you forget to declare \"my $cmd\"?) at Porting/makerel line 115.\nGlobal symbol \"$relroot\" requires explicit package name (did you forget to declare \"my $relroot\"?) at Porting/makerel line 115.\nGlobal symbol \"$reldir\" requires explicit package name (did you forget to declare \"my $reldir\"?) at Porting/makerel line 115.\nGlobal symbol \"$cmd\" requires explicit package name (did you forget to declare \"my $cmd\"?) at Porting/makerel line 116.\nGlobal symbol \"$cmd\" requires explicit package name (did you forget to declare \"my $cmd\"?) at Porting/makerel line 117.\nGlobal symbol \"$relroot\" requires explicit package name (did you forget to declare \"my $relroot\"?) at Porting/makerel line 120.\nGlobal symbol \"$reldir\" requires explicit package name (did you forget to declare \"my $reldir\"?) at Porting/makerel line 120.\nBEGIN not safe after errors--compilation aborted at Porting/makerel line 126.\n"
# expected "Porting/makerel syntax OK\n"
# when executing perl with '-c Porting/makerel'
# Failed test 55 - Porting/valgrindpp.pl compiles at porting/utils.t line 85
#      got "Bareword found where operator expected at Porting/valgrindpp.pl line 52, near \"new IO::File\"\n\t(Do you need to predeclare new?)\nString found where operator expected at Porting/valgrindpp.pl line 52, near \"IO::File \">$opt{\'output-file\'}\"\"\n\t(Do you need to predeclare IO::File?)\nBareword found where operator expected at Porting/valgrindpp.pl line 200, near \"new IO::File\"\n\t(Do you need to predeclare new?)\nScalar found where operator expected at Porting/valgrindpp.pl line 200, near \"IO::File $_\"\n\t(Do you need to predeclare IO::File?)\nsyntax error at Porting/valgrindpp.pl line 52, near \"new IO::File \"\nsyntax error at Porting/valgrindpp.pl line 200, near \"new IO::File \"\nPorting/valgrindpp.pl had compilation errors.\n"
# expected "Porting/valgrindpp.pl syntax OK\n"
# when executing perl with '-c Porting/valgrindpp.pl'
# Failed test 58 - cpan/Pod-Usage/blib/script/pod2usage compiles at porting/utils.t line 85
#      got "Name \"Pod::Usage::Formatter\" used only once: possible typo at cpan/Pod-Usage/blib/script/pod2usage line 135.\ncpan/Pod-Usage/blib/script/pod2usage syntax OK\n"
# expected "cpan/Pod-Usage/blib/script/pod2usage syntax OK\n"
# when executing perl with '-c cpan/Pod-Usage/blib/script/pod2usage'
porting/utils.t .................................................... 
Failed 6/83 subtests 
	(less 3 skipped subtests: 74 okay)

```
These results were generated on an unthreaded build on linux.  The results might differ significantly if run on a threaded build.  There have been subsequent commits to this branch which may have impacted these results.  YMMV.

**Comment by atoomic** [comment]
`ExtUtils-Constant/t/Constant.t`  is fixed by https://github.com/Perl/perl5/pull/17967
cherry-picked 94b5fc74c3d7ed835e83bac40632962af32f5b30 via 66ddf58dc3d88d1f246abaae8c781d561825b2db

**Comment by atoomic** [comment]
porting/cmp_version.t fixed by e4255566

**Comment by jkeenan** [comment]
> porting/cmp_version.t fixed by [e425556](https://github.com/atoomic/perl/commit/e42555664ac5d9112fae9d0129f9e399cf119ab0)

I actually should have closed this ticket once we got all tests running GREEN for objective 1 and merged that branch into `alpha`.  But it appears that once we got into Objective 2 work, the test file (not surprisingly) broke again.

Thank  you very much.
Jim Keenan
